dataloader.py

```diff
@@ -119,15 +138,15 @@ def load(self, buffer_size: int = None) -> None:
         """
         # if unset, set buffersize to number of samples
         if not buffer_size:
-            buffer_size = self.dataset_input.cardinality().numpy()
+            buffer_size = self._dataset_input.cardinality().numpy()
```

Comment:
Missed this one. Should be changed to
`self.n_samples`
I am not sure if this will conflict if
`batch_size > n_samples`